Content.Server/Speech/Components/DamagedSiliconAccentComponent.cs

```diff
@@ -0,0 +1,90 @@
+﻿using Content.Shared.FixedPoint;
+namespace Content.Server.Speech.Components;
+
+[RegisterComponent]
+public sealed partial class DamagedSiliconAccentComponent : Component
+{
+    /// <summary>
+    ///     Enable damage corruption effects
+    /// </summary>
+    [DataField]
+    public bool EnableDamageCorruption = true;
+
+    /// <summary>
+    ///     Override total damage for damage corruption effects
+    /// </summary>
+    [DataField]
+    public bool OverrideTotalDamage = false;
+
+    /// <summary>
+    ///     Value of total damage to use for damage corruption effects if <see cref="OverrideTotalDamage" /> is true.
+    /// </summary>
+    [DataField]
+    public FixedPoint2 OverriddenTotalDamageValue = 0;
+
+    /// <summary>
+    ///     The probability that a character will be corrupted when total damage at or above <see cref="MaxDamageCorruption" />.
+    /// </summary>
+    [DataField]
+    public float MaxDamageCorruption = 0.5f;
+
+    /// <summary>
+    ///     Probability of character corruption will increase linearly to <see cref="MaxDamageCorruption" /> once until
+    ///     total damage is at or above this value.
+    /// </summary>
+    [DataField]
+    public FixedPoint2 DamageAtMaxCorruption = 300;
+
+    /// <summary>
+    ///     Enable charge level corruption effects
+    /// </summary>
+    [DataField]
+    public bool EnableChargeCorruption = true;
+
+    /// <summary>
+    ///     Override charge level for charge level corruption effects
+    /// </summary>
+    [DataField]
+    public bool OverrideChargeLevel = false;
+
+    /// <summary>
+    ///     Value of charge level to use for charge level corruption effects if <see cref="OverrideChargeLevel" /> is true.
+    /// </summary>
+    [DataField]
+    public float OverriddenChargeLevel = 0.0f;
+
+
```

Comment:
Nullable does the same thing.

Reply:
not sure why I didn't think to use nullables as a datafield...